src/libecalc/common/utils/rates.py

```diff
@@ -152,11 +152,28 @@ class Config:
         extra = Extra.forbid
 
     @validator("values", each_item=True, pre=True)
-    def convert_none_to_nan(cls, v: float, field: ModelField) -> TimeSeriesValue:
+    def convert_none_to_nan(cls, v: float, values, field: ModelField) -> TimeSeriesValue:
         if field.outer_type_ is float and v is None:
             return math.nan
         return v
 
+    @root_validator(pre=True)
+    def timesteps_values_one_to_one(cls, field_values):
+        nr_timesteps = len(field_values["timesteps"])
+        nr_values = len(field_values["values"])
+
+        if not cls.__name__ == TimeSeriesVolumes.__name__:
+            if nr_timesteps != nr_values:
+                if all(math.isnan(i) for i in field_values["values"]):
```

Comment:
Some situations: number of timesteps (e.g. 12) differ from number of timesteps values (e.g 2). The values are all NaN. To make things run, the values list is extended with NaN to have same dimension as the timesteps. However, this is probably not the right place to do this. And, ruff fails saying it is a pointless comparison.

Reply:
Done (another PR)